spec/index.bs

```diff
@@ -2282,6 +2282,46 @@ dictionary GPUImageBitmapCopyView {
 
   * {{GPUImageBitmapCopyView/origin}}: If unspecified, defaults to `[0, 0]`.
 
+### <dfn method for=GPUCommandEncoder>copyBufferToBuffer(source, sourceOffset, destination, destinationOffset, size)</dfn> ### {#GPUCommandEncoder-copyBufferToBuffer}
+
+<div algorithm="GPUCommandEncoder.copyBufferToBuffer">
+
+  **Arguments:**
+    - {{GPUBuffer}} |source|
+    - {{GPUSize64}} |sourceOffset|
+    - {{GPUBuffer}} |destination|
+    - {{GPUSize64}} |destinationOffset|
+    - {{GPUSize64}} |size|
+
+  **Returns:** void
+
+  Encode a command into the {{GPUCommandEncoder}} that copies |size| bytes of data from the |sourceOffset| of a {{GPUBuffer}} |source| to the |destinationOffset| of another {{GPUBuffer}} |destination|.
+
+</div>
+
+<div algorithm class=validusage>
+
+  <dfn abstract-op>copyBufferToBuffer Valid Usage</dfn>
+
+  Given a {{GPUCommandEncoder}} |encoder| and the arguments {{GPUBuffer}} |source|, {{GPUSize64}} |sourceOffset|, {{GPUBuffer}} |destination|, {{GPUSize64}} |destinationOffset|, {{GPUSize64}} |size|, the following validation rules apply:
+
+  - |encoder|.{{GPUCommandEncoder/copyBufferToBuffer()}} must be called before calling |encoder|.{{GPUCommandEncoder/finish()}}. // TODO: define the state machine for GPUCommandEncoder.
+  - |encoder|.{{GPUCommandEncoder/copyBufferToBuffer()}} must be called outside of any {{GPURenderPassEncoder}}s.
```

Comment:
I don't really like "outside" here unless we also define "inside", is this more clear? (Ditto for the next bullet)

```suggestion
  - |encoder|.{{GPUCommandEncoder/copyBufferToBuffer()}} must not be called while a {{GPURenderPassEncoder}} is active on |encoder|.
```

Reply:
I like the suggestion. But in the long run we will formally define a state machine (e.g. 'encoding', 'render-encoding', 'compute-encoding', 'finished') and reword this in terms  of that.

Reply:
Yes, and for now we can say that neither a render or compute pass encoder has to be active

Reply:
Done